docs/developer-guide/extensions/proxy-extensions.md

```diff
@@ -0,0 +1,258 @@
+
+## Configuration
+
+As proxy extension is in [Alpha][1] phase, the feature is disabled by
+default. To enable it, it is necessary to configure the feature flag
+in Argo CD command parameters. The easiest way to to properly enable
+this feature flag is by adding the `server.enable.proxy.extension` key
+in the existing `argocd-cmd-params-cm`. For example:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cmd-params-cm
+  namespace: argocd
+data:
+  server.enable.proxy.extension: "true"
+```
+
+Once the proxy extension is enabled, it can be configured in the main
+Argo CD configmap ([argocd-cm][2]).
+
+The example below demonstrate all possible configurations available
+for proxy extensions:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cm
+  namespace: argocd
+data:
+  extension.config: |
+    extensions:
+    - name: httpbin
+      backend:
+        connectionTimeout: 2s
+        keepAlive: 15s
+        idleConnectionTimeout: 60s
+        maxIdleConnections: 30
+        services:
+        - url: http://httpbin.org
+          cluster: https://some-cluster
+```
+
+Every configuration entry is explained below:
+
+#### `extensions` (*list*)
+
+Defines configurations for all extensions enabled.
+
+#### `extensions.name` (*string*)
+(mandatory)
+
+Defines the endpoint that will be used to register the extension
+route. For example, if the value of the property is `extensions.name:
+my-extension` then the backend service will be exposed under the
+following url:
+
+    <argocd-host>/extensions/my-extension
+
+#### `extensions.backend.connectionTimeout` (*duration string*)
+(optional. Default: 2s)
+
+Is the maximum amount of time a dial to the extension server will wait
+for a connect to complete. 
+
+#### `extensions.backend.keepAlive` (*duration string*)
+(optional. Default: 15s)
+
+Specifies the interval between keep-alive probes for an active network
+connection between the API server and the extension server. Optional
+field.
+
+#### `extensions.backend.idleConnectionTimeout` (*duration string*)
+(optional. Default: 60s)
+
+Is the maximum amount of time an idle (keep-alive) connection between
+the API server and the extension server will remain idle before
+closing itself.
+
+#### `extensions.backend.maxIdleConnections` (*int*)
+(optional. Default: 30)
+
+Controls the maximum number of idle (keep-alive) connections between
+the API server and the extension server.
+
+#### `extensions.backend.services` (*list*)
+
+Defines a list with backend url by cluster.
+
+#### `extensions.backend.services.url` (*string*)
+(mandatory)
+
+Is the address where the extension backend must be available.
+
+#### `extensions.backend.services.cluster` (*string*)
+(optional)
+
+If provided, will have to match the application destination name to
+have requests properly forwarded to this service URL. If there are
+multiple backends for the same extension this field is required. The
```

Comment:
Do we actually enforce that this field is set if there's more than one backend service configured?

Reply:
we do:
https://github.com/argoproj/argo-cd/blob/3d02e013391c71a56cc99282d5140b95de2a4e50/server/extension/extension.go#L187